File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numsDup = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int current = nums[i];

            if (numsDup.containsKey(current)) {
                return new int[] {numsDup.get(current), i};
            }

            int diff = target - current;
            numsDup.put(diff, i);
        }
        return new int[2];
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
import math
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0 
        r = len(nums) - 1
        while r >= l:
            currIndex = (r + l) // 2
            if target < nums[currIndex]:
                r = currIndex - 1
            elif target >  nums[currIndex]:
                l = currIndex + 1
            elif target == nums[currIndex]:
                return currIndex
        return -1